### PR TITLE
Add max send buffer per stream option

### PR DESCRIPTION
**Pull request by seanmonstar**
An alternative to #577.

This makes the max send buffer size **per stream**, instead of for the entire connection. It also does a lot less to try to enforce it, since the assigning of capacity is a complicated beast. All this does is clamp how high something like `SendStream::capacity()` will return. It's more just an advisory to the user of the `SendStream`.

Since hyper currently will only produce more data once the available capacity is over zero, it should behave well with this change. I've tried out this patch in the hyper HTTP/2 benchmarks, and see that they all still pass (don't hang). However, this is missing some unit tests in `flow_control.rs`...

### CODE REVIEW

File: src/proto/streams/send.rs [hunk 1]
```diff
@@ -339,7 +343,7 @@ impl Send {
         if available as usize <= buffered {
             0
         } else {
-            available - buffered as WindowSize
+            available.min(self.max_buffer_size as WindowSize) - buffered as WindowSize
```
Comment (nox) [inline src/proto/streams/send.rs:346]: I think this can underflow, right?

Comment (nox) [reply src/proto/streams/send.rs:346]: Never mind, clearly I had not woken up correctly yet.

Comment (seanmonstar) [reply src/proto/streams/send.rs]: No, you were right. I add a test, and it did panic with an overflow. I've fixed that.

File: src/proto/streams/send.rs [hunk 2]
```diff
@@ -340,10 +340,18 @@ impl Send {
...
+            ?buffered,
+            max_buffer_size = self.max_buffer_size
+        );
+
         if available as usize <= buffered {
             0
         } else {
-            available.min(self.max_buffer_size as WindowSize) - buffered as WindowSize
+            available
+                .min(self.max_buffer_size as WindowSize)
```
Comment (nox) [inline src/proto/streams/send.rs:353]: AFAIK this is still not good: `self.max_buffer_size` could be bigger than `u32::MAX`.

File: src/proto/streams/send.rs [hunk 3]
```diff
@@ -340,10 +340,18 @@ impl Send {
         let available = stream.send_flow.available().as_size();
         let buffered = stream.buffered_send_data;
 
+        tracing::info!(
```
Comment (nox) [inline src/proto/streams/send.rs:343]: `info` is bigger than `debug` right? I would use `debug` or lower here.
